### PR TITLE
Stactools Package

### PR DESCRIPTION
**Pull request by pholleway**
**Related Issue(s):**
Initial src and test file set up. 

**Description:**
Included set up for contants.py and stac.py

**PR checklist:**

- [x] Code is formatted (run `scripts/format`).
- [x] Code lints properly (run `scripts/lint`).
- [x] Tests pass (run `scripts/test`).
- [x] Documentation has been updated to reflect changes, if applicable.
- [ ] Changes are added to the [CHANGELOG](../CHANGELOG.md).
- [ ] Examples are added to `examples/`

### CODE REVIEW

**Comment by pholleway** [comment]
cc @gadomski

**Comment by gadomski** [comment]
@pholleway _should_ be updated and hopefully CI passes. FYI you'll need to run the following to set up the new linting/formatting infrastructure in your virtual environment:

```shell
$ pip install -r requirements-dev.txt && pre-commit install
```

File: docs/source/conf.py [hunk 1]
```diff
@@ -14,39 +14,39 @@
...
 
+from typing import List
 
 # -- Project information -----------------------------------------------------
 
-project = 'drcog-lulc'
-copyright = '2022, Paola Holleway'
-author = 'Paola Holleway'
+project = "drcog-lulc"
+copyright = "2022, Paola Holleway"
```
Comment (gadomski) [inline docs/source/conf.py:22]: This copyright should be Microsoft.

File: src/stactools/drcog_lulc/constants.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    Summaries,
+    TemporalExtent,
+)
+from pystac.extensions.item_assets import AssetDefinition
+
+ITEM_ID = "drcog-hrlulc-pilot"
+ITEM_DESCRIPTION = "DRCOG LULC at 1m resolution for year 2018"
+# Per the discussion in https://github.com/radiantearth/stac-spec/issues/216,
+# the recommendation for multi-platform datasets is to include all platforms
+# and use a string separator. The same logic is applied to the mission.
```
Comment (gadomski) [inline src/stactools/drcog_lulc/constants.py:20]: This comment can be removed, this was for a different use-case where we had multiple missions.

File: src/stactools/drcog_lulc/constants.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+            {"value": 8, "description": "Cropland", "color-hint": "FFAE42"},
+        ],
+    }
+}
+
+COLLECTION_ID = "drcog-hrlulc"
+COLLECTION_TITLE = "DRCOG LULC 2018"
+COLLECTION_DESCRIPTION = (
+    "1,000 square miles were classified by Land Use Land Cover at 1-meter resolution"
+)
```
Comment (gadomski) [inline src/stactools/drcog_lulc/constants.py:63]: Can we put more information in the collection description? E.g. where the data were derived from, who owns the data, etc?

File: src/stactools/drcog_lulc/constants.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+LICENSE = "proprietary"
+LICENSE_LINK = Link(
+    rel="license", target="https://drcog.org/legal-terms", title="DRCOG Legal Terms"
+)
+KEYWORDS = ["Land Cover", "Land Use", "NAIP", "USDA"]
+PROVIDERS = [
+    Provider(
+        name="DRCOG",
```
Comment (gadomski) [inline src/stactools/drcog_lulc/constants.py:71]: I think we should use the full name here, e.g. "Denver Regional Council of Governments"

File: src/stactools/drcog_lulc/constants.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+            "description": ASSET_PROPS["data"]["description"],
+            "raster:bands": ASSET_PROPS["data"]["bands"],
+            "classification:classes": ASSET_PROPS["data"]["classes"],
+            "roles": ASSET_PROPS["data"]["roles"],
+        }
+    )
+}
+REPORT_LINK = Link(
+    rel="describedby",
+    target="https://gis.drcog.org//rdc/supplemental/lulc_pilot_report.zip",  # noqa
```
Comment (gadomski) [inline src/stactools/drcog_lulc/constants.py:103]:
```suggestion
    target="https://gis.drcog.org/rdc/supplemental/lulc_pilot_report.zip",
```

File: src/stactools/drcog_lulc/stac.py [hunk 6]
```diff
@@ -1,112 +1,83 @@
...
+    DRCOG LULC classification product.
+    Optionally creates an additional Asset for an input quality COG if passed an
+    href.
+    Args:
+        asset_href (str): An href to a COG containing a tile of classication data.
+        read_href_modifier (Callable[[str], str]): An optional function to
+            modify the MTL and USGS STAC hrefs (e.g. to add a token to a url).
     Returns:
-        Collection: STAC Collection object
+        Item: STAC Item object representing the worldcover tile
```
Comment (gadomski) [inline src/stactools/drcog_lulc/stac.py:28]:
```suggestion
        Item: STAC Item object representing the landcover GeoTIFF tile
```

**Review by gadomski — CHANGES_REQUESTED** [review]
A more general note to go with my specific comments: can you add some text to the README describing the process by which the source ESRI-formatted file was converted to GeoTIFF?

File: docs/source/conf.py (hunk 1 above)
Comment (pholleway) [reply docs/source/conf.py:22]: Updated to Microsoft

File: src/stactools/drcog_lulc/constants.py (hunk 2 above)
Comment (pholleway) [reply src/stactools/drcog_lulc/constants.py:20]: Comment removed

File: src/stactools/drcog_lulc/constants.py (hunk 4 above)
Comment (pholleway) [reply src/stactools/drcog_lulc/constants.py:71]: Updated to full name.

File: src/stactools/drcog_lulc/constants.py (hunk 3 above)
Comment (pholleway) [reply src/stactools/drcog_lulc/constants.py:63]: Added more information - owner/partnerships and data inputs.

File: src/stactools/drcog_lulc/constants.py (hunk 2 above)
Comment (gadomski) [reply src/stactools/drcog_lulc/constants.py:20]:
```suggestion
```

File: src/stactools/drcog_lulc/constants.py (hunk 3 above)
Comment (gadomski) [reply src/stactools/drcog_lulc/constants.py:63]: I don't see these changes.

File: src/stactools/drcog_lulc/constants.py (hunk 4 above)
Comment (gadomski) [reply src/stactools/drcog_lulc/constants.py:71]: I don't see these changes.

**Review by gadomski — CHANGES_REQUESTED** [review]

**Review by gadomski — APPROVED** [review]

**Pull request merged**